basic docs for nested column query functions (#12922)
apache/druid: docs/querying/sql-data-types.md

Druid natively supports five basic column types: "long" (64 bit signed int), "float" (32 bit float), "double" (64 bit
float) "string" (UTF-8 encoded strings and string arrays), and "complex" (catch-all for more exotic data types like

> json, hyperUnique, and approxHistogram columns).

Author: not really sure, i felt compelled to add json because i talk about it later on

When `druid.expressions.useStrictBooleans = true`, Druid uses three-valued logic for
[expressions](../misc/math-expr.md) evaluation, such as `expression` virtual columns or `expression` filters.
However, even in this mode, Druid uses two-valued logic for filter types other than `expression`.

## Nested columns

> Druid supports storing nested data structures in segments using the native `COMPLEX<json>` type. This data can be
> interacted with using [JSON functions](sql-json-functions.md), which can extract nested values, parse from string,

Reviewer: ```suggestion
Druid supports storing nested data structures in segments using the native `COMPLEX<json>` type. You can interact
with this data using [JSON functions](sql-json-functions.md), which can extract nested values, parse from string,
```

serialize to string, and to create new `COMPLEX<json>` structures.

`COMPLEX` types in general currently have limited functionality outside of the use of the specialized functions which
understand them, and so have undefined behavior when:
* grouping on complex values
* filtered directly on complex values, e.g. `WHERE json is NULL`
* used as inputs to aggregators without specialized handling for a specific complex type

> In many cases, these functions are provided for translating these `COMPLEX` value types a `STRING`, which serves as

Reviewer: ```suggestion
In many cases, functions are provided to translate `COMPLEX` value types to `STRING`, which serves as a
```
